lib/templates/v_0.4.0/api/server-controllers.js

```diff
@@ -16,43 +16,104 @@ var __Proper__ = require('mongoose').model('__Proper__')
+      } else {
+        res.send({
+          success: true
+          , __name__s: __name__s
+          , pagination: {
+            per: per
+            , page: page
+          }
+        });
+      }
+    });
+  } else {
+    console.log('list __name__s');
+    __Proper__.find({}).exec(function(err, __name__s) {
+      if(err || !__name__s) {
+        res.send({ success: false, message: err });
+      } else {
+        res.send({ success: true, __name__s: __name__s });
+      }
+    });
+  }
 }
 
 exports.search = function(req, res) {
-  // search by query parameters 
+  //search by query parameters
   // up to front end to make sure the params exist on the model
-  console.log('searching for __name__s with params.');
+  console.log("searching for __name__s with params.");
   var mongoQuery = {};
+  var page, per;
   for(key in req.query) {
     if(req.query.hasOwnProperty(key)) {
-      console.log('found query param: ' + key);
-      mongoQuery[key] = req.query[key];
+      if(key == "page") {
+        page = req.query.page;
+      } else if(key == "per") {
+        per = req.query.per;
+      } else {
+        console.log("found search query param: " + key);
+        mongoQuery[key] = req.query[key];
+      }
     }
   }
-  console.log(mongoQuery);
-  __Proper__.find(mongoQuery).exec(function(err, __name__s) {
-    if(err || !__name__s) {
+  if(page || per) {
+    console.log("searching for __name__s with pagination");
+    console.log(mongoQuery);
+    page = page || 1;
+    per = per || 20;
+    __Proper__.find(mongoQuery).skip((page-1)*per).limit(per).exec(function(err, __name__s) {
+      if(err || !__name__s) {
+        res.send({ success: false, message: err });
+      } else {
+        res.send({ 
+          success: true
+          , __name__s: __name__s
+          , pagination: {
+            per: per
+            , page: page
+          }
+        });
+      }
+    });
+  } else {
+    console.log(mongoQuery);
+    __Proper__.find(mongoQuery).exec(function(err, __name__s) {
+      if(err || !__name__s) {
+        res.send({ success: false, message: err });
+      } else {
+        res.send({ success: true, __name__s: __name__s });
+      }
+    });
+  }
+}
+
+exports.getById = function(req, res) {
+  console.log('get __name__ by id');
+  __Proper__.findById(req.params.id).exec(function(err, __name__) {
+    if(err) {
       res.send({ success: false, message: err });
+    } else if (!__name__) {
+      res.send({ success: false, message: "no __name__ found :(" });
     } else {
-      res.send({ success: true, __name__s: __name__s});
+      res.send({ success: true, __name__: __name__ });
     }
   });
 }
 
-exports.getById = function(req, res) {
+exports.getAndPopulate = function(req, res) {
   console.log('get __name__ by id');
-  __Proper__.findOne({ _id: req.params.id }).exec(function(err, __name__) {
+  __Proper__.findById(req.params.id).populate('author').exec(function(err, __name__) {
```

Comment:
let's put this as a commented out line.  99% of all models created in the cli won't have author.  so it could be like this

```javascript
   __Proper__.findById(req.params.id)
        // .populate('author')
        .exec(function(err, __name__) {
```